File: app/src/main/java/com/example/paritosh/stormy/WeatherContract.java
package com.example.paritosh.stormy;


import android.content.Context;
import android.support.annotation.StringRes;

import com.example.paritosh.stormy.model.CurrentWeatherDataBindingModel;
import com.example.paritosh.stormy.model.Hourly;

public interface WeatherContract {

    interface WeatherView {

        void render(CurrentWeatherDataBindingModel model);

        void loadHourlyData(Hourly[] hourlyData);

        void showMessage(@StringRes int messageResId);

        void showErrorMessage(@StringRes int titleResId, @StringRes int messageResId);

        Context loadContext();


    }

    interface WeatherPresenter {
        void updateWeatherDetails();

        void refreshWeatherData();

    }


}

File: app/src/main/java/com/example/paritosh/stormy/WeatherContract.kt
package com.example.paritosh.stormy

import android.content.Context
import com.example.paritosh.stormy.model.CurrentWeatherDataBindingModel

interface WeatherContract {

    interface WeatherView {

        fun render(model: CurrentWeatherDataBindingModel)

        fun showMessage(messageResId: Int)

        fun showErrorMessage(titleResId: Int, messageResId: Int)

        fun loadContext(): Context
    }

    interface WeatherPresenter {

        fun updateWeatherDetails()

        fun refreshWeatherDetails()
    }
}